src/Bravo/Signer.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Signer.h"
+
+#include <TrezorCrypto/ecdsa.h>
+#include <TrezorCrypto/secp256k1.h>
+
+using namespace TW::Bravo;
+
+void Signer::sign(const PrivateKey& privateKey, Transaction& transaction, int(*canonicalChecker)(uint8_t by, uint8_t sig[64])) const {
+    if (!transaction.isValid()) {
+        throw std::invalid_argument("Invalid Transaction!");;
+    }
+
+    Data result;
+    result.resize(65);
+    if (ecdsa_sign_digest(&secp256k1, privateKey.bytes.data(), hash(transaction).data(), result.data() + 1, result.data(), canonicalChecker) != 0) {
+        throw std::runtime_error("Signing failed.");
+    }
+
+    result[0] += 31;
+
+    transaction.addSignature(Signature(result));
+}
+
+
+bool Signer::verify(const PrivateKey& privateKey, const Signature& signature, Transaction& transaction) const noexcept {
```

Comment:
try `PublicKey.verify`

Reply:
I think we are not using `Signer::verify` anymore in test code. I will remove this.